packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -1291,6 +1289,21 @@ export class HardhatNode extends EventEmitter {
     }
   }
 
+  private async _runInPendingBlockContext<T>(action: () => Promise<T>) {
+    const snapshotId = await this.takeSnapshot();
+    let result;
+    try {
+      await this.mineBlock(false);
+      result = await action();
+    } catch (err) {
+      throw err;
+    } finally {
+      await this.revertToSnapshot(snapshotId);
+    }
+
+    return result;
```

Comment:
This can be simplified to:
```suggestion
    try {
      await this.mineBlock(false);
      return await action();
    } finally {
      await this.revertToSnapshot(snapshotId);
    }
```
Btw, we usually don't `await` on the returned promise. This situation is an exception to this rule. Do you know why `return await` is necessary here? 😀

Reply:
Woah. Really interesting find. I did some research and found an [article](https://jakearchibald.com/2017/await-vs-return-vs-return-await/) about it made by our favorite JS guy - Jake Archibald. I never knew you could do something like that.

Now on the other hand... Is it a good idea to return `undefined` if `_runInPendingBlockContext` fails? 🤔